### PR TITLE
988 bcs in tutorials

### PR DESCRIPTION
**Pull request by IngridKJ**
## Proposed changes

After merging of #950 there was a lack of clarity on how to set boundary conditions. This PR introduces a tutorial for setting boundary conditions for the scalar problem and vector problem, as well as time dependent boundary conditions. Credits to @Yuriyzabegaev 

## Types of changes

- [ ] Minor change (e.g., dependency bumps, broken links).
- [ ] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Testing (contribution related to testing of existing or new functionality).
- [ ] Documentation (contribution related to adding, improving, or fixing documentation).
- [ ] Maintenance (e.g., improve logic and performance, remove obsolete code).
- [x] Other: Tutorials

## Checklist

- [x] The documentation is up-to-date.
- [ ] Static typing is included in the update.
- [x] This PR does not duplicate existing functionality.
- [ ] The update is covered by the test suite (including tests added in the PR).
- [ ] If new skipped tests have been introduced in this PR, `pytest` was run with the `--run-skipped` flag.

### CODE REVIEW

File: tutorials/boundary_conditions.ipynb [hunk 1]
```diff
@@ -0,0 +1,439 @@
+{
```
Comment (IvarStefansson) [inline tutorials/boundary_conditions.ipynb:1]: <p>Could we say "Before reading this tutorial, we recommend reading <em>the single-phase flow tutorial."?</em></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1073/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (IvarStefansson) [inline tutorials/boundary_conditions.ipynb:1]: <p>Please explain that bc_type_-methods are only defined for fluxes. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1073/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (IvarStefansson) [inline tutorials/boundary_conditions.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #28.            pressure_vals[domain_sides.east] = 5 * self.units.Pa
</pre><p>Please use self.fluid.convert_units(5, "Pa"), which divides by the Pascal value.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1073/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (IvarStefansson) [inline tutorials/boundary_conditions.ipynb:1]: <p>End of case 1:</p><p>"We used this type of boundary condition on all but the east boundary in the example above."</p><p>Case 2:</p><p>The boundary values for the advected quantity will be computed automatically using the same constitutive laws as in the interior.</p><p>End of case 3:</p><p>"We used this type of boundary condition on the east boundary in the example above."</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1073/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (IvarStefansson) [inline tutorials/boundary_conditions.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #26.            darcy_flux_vals[influx_cells] = -1 * self.units.Pa / self.units.m
</pre><p>Use convert_units</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1073/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (IvarStefansson) [inline tutorials/boundary_conditions.ipynb:1]: <p>I would rename the section to Momentum balance and skip the sentence "In the case of the mechanics problem we utilize the momentum balance model class. "</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1073/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (IvarStefansson) [inline tutorials/boundary_conditions.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #23.            values = np.ones((self.nd, bg.num_cells))
</pre><p>Are you sure you want to set ones on east and west?</p><p>For consistency with the above examples, please scale all values assigned here.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1073/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (IvarStefansson) [inline tutorials/boundary_conditions.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #32.            return values.ravel("F")
</pre><p>Could you please add a comment about the "F" convention in PorePy?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1073/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (IvarStefansson) [inline tutorials/boundary_conditions.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #47.            values = np.ones((self.nd, bg.num_cells))
</pre><p>I think in general it's safer to initialize with zeros.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1073/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

File: tutorials/single_phase_flow.ipynb [hunk 2]
```diff
@@ -183,9 +183,17 @@
       "\n"
```
Comment (IvarStefansson) [inline tutorials/single_phase_flow.ipynb]: <p>Thanks!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1073/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

**Comment by IvarStefansson** [comment]
Great work! Left some comments, let me know if you want to discuss anything.

File: tutorials/boundary_conditions.ipynb (hunk 1 above)
Comment (Yuriyzabegaev) [reply tutorials/boundary_conditions.ipynb:1]: <p>To set a boundary condition do the following: </p><p>1 ...... bc_type_"flux_name" &lt;- ((Needs more explanation. Somewhere.)) Point is that we do not have a bc_type_pressure, which might be confusing if it is not mentioned. </p><p>2 ...... bc_values_"physical_quantity"</p><p> </p><p>bc_type_mechanics -&gt; bc_type_stress, keep the old one and throw an error (as help to users such that they rename it). Will be separate PR.</p>

File: tutorials/boundary_conditions.ipynb [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
```
Comment (IvarStefansson) [inline tutorials/boundary_conditions.ipynb:1]: <p>I don't think this is exactly what I understood to be the plan. That's fine, but I suggest:</p><p> </p><p>To set a boundary condition do the following:</p><p>1) Specify the type of the boundary condition (Dirichlet, Neumann or Robin) in the method named <code>bc_type_&lt;flux_name&gt;</code>.</p><p>2) Set the values of relevant physical quantities on the boundary in the method named <code>bc_values_&lt;physical_quantity&gt;</code>. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1073/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (IvarStefansson) [inline tutorials/boundary_conditions.ipynb:1]: <p>Please add "a":</p><p>We can directly set the values of F on a boundary by:</p><p> </p><p>Could we change the order of sentences starting with "Pro tip" and "We used...". I think that makes it clearer what "this type" refers to.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pmgbergen/porepy/pull/1073/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

**Review by IvarStefansson — APPROVED** [review]
Thanks [alot](http://hyperboleandahalf.blogspot.com/2010/04/alot-is-better-than-you-at-everything.html)!

**Pull request merged**